libsolidity/interface/Natspec.cpp

```diff
@@ -110,9 +124,25 @@ Json::Value Natspec::devDocumentation(ContractDefinition const& _contractDef)
 				methods[it.second->externalSignature()] = method;
 			}
 		}
+		if (auto var = dynamic_cast<VariableDeclaration const*>(&it.second->declaration()))
+		{
+			Json::Value declarationDoc(devDocumentation(var->annotation().docTags));
+			if (!declarationDoc.empty() && var->isStateVariable() && var->isPublic())
+				methods[it.second->externalSignature()] = declarationDoc;
+		}
+	}
+
+	Json::Value stateVariables(Json::objectValue);
+	for (auto const& it: _contractDef.stateVariables())
+	{
+		Json::Value documentation(devDocumentation(it->annotation().docTags));
+		if (!documentation.empty())
+			stateVariables[it->name()] = documentation;
```

Comment:
```cpp
    for (VariableDeclaration const* varDecl: _contractDef.stateVariables())
        if (auto const doc = devDocumentation(varDecl->annotation().docTags); !doc.empty())
            stateVariables[varDecl->name()] = doc;
```

That's my multiline code change suggestion.

* Use explicit type in (range based) for loop
* shortening the loop body.
* use const var if var is not meant to be modified.